### PR TITLE
fix gradient dg

### PR DESCRIPTION
**Pull request by JustusWerner**
Dear julia-github co-worker,

I am a bachelor student from Germany. While working on my bachelor thesis I got in touch with Julia and its sensitivityanalysis.
Your documentary did help to understand how adjoint_sensitivities() works but it is not beginner-friendly and 
at the very least it is wrong regarding the gradient dg.
Moreover, I am keen in contributing more to the descriptive part of this page if such a contribution is welcome. I would like
to add an explanation of the mathematical background according to your example to help make the paragraph easier to understand. 

Yours sincerely
Justus Werner

### CODE REVIEW

File: docs/src/analysis/sensitivity.md [hunk 1]
```diff
@@ -604,21 +604,22 @@ In this case we'd like to calculate the adjoint sensitivity of the scalar energy
...
 
 Notice that the gradient of this function with respect to the state `u` is:
 
 ```julia
 function dg(out,u,p,t)
-  out[1]= u[1] + u[2]
-  out[2]= u[1] + u[2]
+  out[1]= u[1] 
+  out[2]= u[2]
+  out[3]= u[3]
```
Comment (YingboMa) [inline docs/src/analysis/sensitivity.md]: I believe that this section is for the Lotka Volterra model, so there are only two elements.

File: docs/src/analysis/sensitivity.md [hunk 2]
```diff
@@ -604,21 +604,22 @@ In this case we'd like to calculate the adjoint sensitivity of the scalar energy
 functional
 
 ```math
-G(u,p)=\int_{0}^{T}\frac{\sum_{i=1}^{n}u_{i}^{2}(t)}{2}dt
+G(u,p)=\int_{0}^{T}\frac{\sum_{i=1}^{3}u_{i}^{2}(t)}{2}dt
```
Comment (YingboMa) [inline docs/src/analysis/sensitivity.md]: This isn't necessary.

File: docs/src/analysis/sensitivity.md [hunk 3]
```diff
@@ -604,21 +604,22 @@ In this case we'd like to calculate the adjoint sensitivity of the scalar energy
...
 ```
 
 which is
 
 ```julia
-g(u,p,t) = (sum(u).^2) ./ 2
+g(u,p,t) = (sum(u).^2) / 2
```
Comment (YingboMa) [inline docs/src/analysis/sensitivity.md]: `sum(u).^2` -> `sum(u)^2`

File: docs/src/analysis/sensitivity.md [hunk 4]
```diff
@@ -636,7 +637,7 @@ differentiation as follows:
 function G(p)
   tmp_prob = remake(prob,p=p)
   sol = solve(tmp_prob,Vern9(),abstol=1e-14,reltol=1e-14)
-  res,err = quadgk((t)-> (sum(sol(t)).^2)./2,0.0,10.0,abstol=1e-14,reltol=1e-10)
+  res,err = quadgk((t)-> (sum(sol(t)).^2)/2,0.0,10.0,abstol=1e-14,reltol=1e-10)
```
Comment (YingboMa) [inline docs/src/analysis/sensitivity.md]: Same here

File: docs/src/analysis/sensitivity.md (hunk 1 above)
Comment (JustusWerner) [reply docs/src/analysis/sensitivity.md]: This section is the gradient of the function g above and not regarding to any ODE.

File: docs/src/analysis/sensitivity.md (hunk 3 above)
Comment (JustusWerner) [reply docs/src/analysis/sensitivity.md]: To rebuild the function under the integral in G you even need  
sum(u.^2)/2 
Sorry there, my bad.

File: docs/src/analysis/sensitivity.md (hunk 4 above)
Comment (JustusWerner) [reply docs/src/analysis/sensitivity.md]: Here it is correct but the ./ is unecessary

**Comment by ChrisRackauckas** [comment]
I'm closing this since other PRs like https://github.com/JuliaDiffEq/DiffEqDocs.jl/pull/318 are taking this on, but @hpodhaisky might want to take note of this!

**Comment by ChrisRackauckas** [comment]
Thank you for your contribution. It is helping us finally push forward our examples.

**Pull request closed without merging**